pkg/distributor/ratestore_test.go

```diff
@@ -156,6 +156,33 @@ func TestRateStore(t *testing.T) {
 		time.Sleep(time.Second)
 		require.Equal(t, int64(0), tc.rateStore.RateFor("tenant 1", 0))
 	})
+
+	t.Run("it clears the rate after an interval", func(t *testing.T) {
+		tc := setup(true)
+		tc.ring.replicationSet = ring.ReplicationSet{
+			Instances: []ring.InstanceDesc{
+				{Addr: "ingester0"},
+			},
+		}
+
+		tc.clientPool.clients = map[string]client.PoolClient{
+			"ingester0": newRateClient([]*logproto.StreamRate{
+				{Tenant: "tenant 1", StreamHash: 1, StreamHashNoShard: 0, Rate: 25},
+			}, 1),
+		}
+
+		tc.rateStore.rateKeepAlive = 2 * time.Second
+		_ = tc.rateStore.StartAsync(context.Background())
+		defer tc.rateStore.StopAsync()
+
+		require.Eventually(t, func() bool {
+			return 25 == tc.rateStore.RateFor("tenant 1", 0)
+		}, time.Second, time.Millisecond)
```

Comment:
nit: checking every millisecond with a 1 second sounds too much imo
```suggestion
		}, time.Second, 100*time.Millisecond)
```